Test/Makefile

```
@@ -9,9 +9,17 @@
 
 # Do we perform comparisons between the actual outputs and expected
 # outputs now (and thus fail as soon as there is a difference), or
-# not (in which case user is expected to compare them after we're
-# done building them here)?
-DIFFNOW=1			# 1 = do diff now; 0 = don't do diffs now
+# aftwer we are done with everything else?
+DIFFEND=0			# 0 = do diff now; 1 = do diffs at end of processing
+
+DIFFSWITCHES=" "		# switches user can pass in to # override — not yet implemented
+S4D1=-C0 -bBw			# switches used for most calls to diff
+S4DR=-C0 -bBw -r		# switches used for recursive diff
+
+DIFFOUT=./DIFFALL.bash	# path to output program to be run (iff DIFFEND=1)
```

I see that you already updated this line, but here's yet another proposal:
Using `mktemp` will actually create the file (not only set its name) and return its filename (absolute path). This will make sure the file exists.
```suggestion
DIFFOUT!=mktemp	# path to output program to be run (iff DIFFEND=1)
```

---

Yeah, I am a bit of a fan of `mktemp` (I say “bit of” because in theory I like it, but in practice I always have to re-read the man page; if I were _really_ a fan, I wouldn’t need to do that : -), but if we use a dynamically generated name, we have no mechanism for having `make clean` remove the temp file. (Or of putting it in .gitignore, as per your next comment.)

But, he said thinking aloud, what if we used
~~~make
DIFFOUT := $(mktemp TestMakefileTempXXX.erasethistempfile)
~~~
we could then have `make clean` perform `rm *.erasethistempfile` and add `*.erasethistempfile` to .gitignore, no?

---

isn't it enough to do `rm $(DIFFOUT)` for removing the tempfile again?

Also, a dynamically created file by `mktemp` usually is not created in the current working directory so we don't need to exclude it in `.gitignore`.  That said, there's nothing against providing the suffix  (to the `mktemp` command) and excluding this suffix in `.gitignore`

---

Well, if we put `rm $(DIFFOUT)` in this target, the user would not be able to easily re-run the comparisons, which could be quite useful. The idea is that `make clean` should remove that file. Thus it has to be remembered across invocations of `make`, which is why I shy away from `mktemp`. (But using `--tmpdir=.` would solve the not-in-this-directory part, but not the name part.)

Which raises the point that it would be a good idea to have this target end by telling user how to do that; something like “if you want to repeat the comparisons only, issue ./$(DIFFOUT) from the Stylesheets/Test/ directory”.